### PR TITLE
Update AirLift for Boards and Breakouts

### PR DESCRIPTION
**Pull request by brentru**
This is a **breaking** PR for WINC1500 users!

Changes:
* Updated `AdafruitIO_WiFi` for `AdafruitIO_AIRLIFT`
* Added configurable pins to `AdafruitIO_AIRLIFT` for using with breakouts and boards
* Added configurable pins to WINC1500
* Bumped version to 3.1.0

**Usage with AirLift**
```
#define USE_AIRLIFT
AdafruitIO_WiFi io(IO_USERNAME, IO_KEY, WIFI_SSID, WIFI_PASS, SPIWIFI_SS, SPIWIFI_ACK, ESP32_RESETN, ESP32_GPIO0);
```

_Usage with WINC1500_
```
#define USE_WINC1500
AdafruitIO_WiFi io(IO_USERNAME, IO_KEY, WIFI_SSID, WIFI_PASS, WINC_CS, WINC_IRQ, WINC_RST, WINC_EN);
```

This PR has been tested with a PyPortal and a Feather M0 + AirLift ESP32 Breakout

### CODE REVIEW

File: examples/adafruitio_00_publish/config.h [hunk 1]
```diff
@@ -13,24 +13,26 @@
...
 
 // comment out the following two lines if you are using fona or ethernet
 #include "AdafruitIO_WiFi.h"
+
+// if you are using ESP32, MKR1000, ESP8266 or WICED, use the following line
 AdafruitIO_WiFi io(IO_USERNAME, IO_KEY, WIFI_SSID, WIFI_PASS);
 
-/******************************* AirLift **************************************/
+// uncomment the following two lines if you are using airlift,
+// #define USE_AIRLIFT
```
Comment (ladyada) [inline examples/adafruitio_00_publish/config.h]: this has to be before the #include "AdafruitIO_WiFi.h" (?)

File: examples/adafruitio_00_publish/config.h [hunk 2]
```diff
@@ -13,24 +13,26 @@
...
-// the AdafruitIO_AirLift client will work with the following boards:
-//   - Adafruit PyPortal -> https://www.adafruit.com/product/4116
-//   - Adafruit Metro M4 Express AirLift Lite -> https://www.adafruit.com/product/4000
-
-// uncomment the following two lines for AirLift,
-// and comment out the AdafruitIO_WiFi client in the WIFI section
-// #include "AdafruitIO_AIRLIFT.h"
-// AdafruitIO_AIRLIFT io(IO_USERNAME, IO_KEY, WIFI_SSID, WIFI_PASS);
+// uncomment the following two lines if you are using winc1500,
+// #define USE_WINC1500
```
Comment (ladyada) [inline examples/adafruitio_00_publish/config.h]: ditto

File: src/AdafruitIO_Definitions.h [hunk 3]
```diff
@@ -42,9 +42,9 @@ class AdafruitIOGroupCallback {
 };
 
 // Uncomment/comment to turn on/off debug output messages.
-// #define AIO_DEBUG
+#define AIO_DEBUG
```
Comment (ladyada) [inline src/AdafruitIO_Definitions.h]: i think you want to remove before release?

File: src/wifi/AdafruitIO_AIRLIFT.h [hunk 4]
```diff
@@ -26,13 +26,7 @@
...
 #include "Adafruit_MQTT_Client.h"
 
-// Configure some pins used for the ESP32 connection
-#if !defined(SPIWIFI_SS) // if the wifi definition isnt in the board variant
-  #define BOARDEF 1
-  #define SPIWIFI SPI
-  #define ESP32_GPIO0 -1 // Not connected
-#endif
-
+#define SPIWIFI SPI
```
Comment (ladyada) [inline src/wifi/AdafruitIO_AIRLIFT.h]: we'll need to have to have this be configurable with the setpins FYI

File: src/wifi/AdafruitIO_AIRLIFT.h [hunk 5]
```diff
@@ -165,10 +154,15 @@ class AdafruitIO_AIRLIFT : public AdafruitIO {
     /**************************************************************************/
     void _connect()
     {
+      // setup ESP32 pins
+      if (_ssPin != 0) {
```
Comment (ladyada) [inline src/wifi/AdafruitIO_AIRLIFT.h]: pin 0 is a valid pin, use -1 for 'unused' pins

File: src/wifi/AdafruitIO_WINC1500.h [hunk 6]
```diff
@@ -21,29 +21,77 @@
...
-#define WINC_IRQ  7
-#define WINC_RST  4
-#define WINC_EN   2
-
 class AdafruitIO_WINC1500 : public AdafruitIO {
 
   public:
-    AdafruitIO_WINC1500(const char *user, const char *key, const char *ssid, const char *pass);
-    ~AdafruitIO_WINC1500();
+    AdafruitIO_WINC1500(const char *user, const char *key, const char *ssid, const char *pass, int winc_cs, int winc_irq, int winc_rst, int winc_en) : AdafruitIO(user, key)
```
Comment (ladyada) [inline src/wifi/AdafruitIO_WINC1500.h]: make the default pins

#define WINC_CS   8
#define WINC_IRQ  7
#define WINC_RST  4
#define WINC_EN   2

File: examples/adafruitio_00_publish/config.h [hunk 7]
```diff
@@ -13,24 +13,27 @@
...
-//   - Adafruit Metro M4 Express AirLift Lite -> https://www.adafruit.com/product/4000
+AdafruitIO_WiFi io(IO_USERNAME, IO_KEY, WIFI_SSID, WIFI_PASS);
 
-// uncomment the following two lines for AirLift,
-// and comment out the AdafruitIO_WiFi client in the WIFI section
-// #include "AdafruitIO_AIRLIFT.h"
-// AdafruitIO_AIRLIFT io(IO_USERNAME, IO_KEY, WIFI_SSID, WIFI_PASS);
+// uncomment the following line if you are using airlift
+// and comment out the AdafruitIO_WiFi client above
+// AdafruitIO_WiFi io(IO_USERNAME, IO_KEY, WIFI_SSID, WIFI_PASS, SPIWIFI_SS, SPIWIFI_ACK, ESP32_RESETN, ESP32_GPIO0, &SPIWIFI);
```
Comment (ladyada) [inline examples/adafruitio_00_publish/config.h]: use #ifdef USE_AIRLIFT to auto select which line

**Comment by brentru** [comment]
@ladyada  OK, change made to `config.h` - tested publish&subscribe with PyPortal, M0 WiFi WINC1500, AirLift Breakout + Feather M0, and Metro M4 WiFi.

**Comment by ladyada** [comment]
ok sounds good!

**Pull request merged**